homeassistant/exceptions.py

```diff
@@ -21,10 +23,18 @@ def __init__(
         translation_placeholders: dict[str, str] | None = None,
     ) -> None:
         """Initialize exception."""
-        super().__init__(*args)
+        # pylint: disable-next=import-outside-toplevel
+        from .helpers.translation import async_get_exception_message
+
         self.translation_domain = translation_domain
         self.translation_key = translation_key
         self.translation_placeholders = translation_placeholders
+        if message := async_get_exception_message(
+            translation_domain, translation_key, translation_placeholders
+        ):
+            self._message = message
+        message_and_args = tuple(arg for arg in (message, *args) if arg is not None)
+        super().__init__(*message_and_args)
```

Comment:
The logic seems backwards.
I think we should generate a translated message if we were not passed any positional arguments, something like this:

```suggestion
        if not args and (
            message := async_get_exception_message(
                translation_domain, translation_key, translation_placeholders
            )
        ):
            args = (message,)
        super().__init__(*args)
```